### ISSUE TITLE
Incorrect Histogram

### ISSUE THREAD

**Issue by waqarmalik** [bokeh/bokeh#4338]
While using the `Histogram` chart, I'm running into problems where the last few bins are incorrectly depicted. Consider the code below, with a hard coded array for which I was able to catch a problem.

```
import numpy as np
from bokeh.charts import Histogram, output_file, show

a = np.array([  0,  89,  60,   0,  62,  76,   0,  62,   0,  70,   0,   0,   0,
        77,   0,  53,  67,  49,   0,   0,  61,   0,  55,  68,   0, 102,
        60,   0,  42,   0,   0,   0,   0,  66,   0,   5,  82,  64,  46,
        61,   0,   0,   0,  96,  89,   0,  74,   0,   0,  45,   0,   0,
        22,  55,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,
         0,   8,   0,   0,   0,   0,   0,   0,  26,   0,  37,   0,   0,
         0,   0,   0,   0,   0,   0,   0,   0,   9,   0,   0,  77,   0,
        61,   0,   0,  15,  12,  53,   0,   0,   0,  73,  68,  75,  71,
        68,  42,  50,  44,  37,  42,  40,  51,  59,  88,  43,  83,  78,
        80,  85,  48,  43,  47,  51,  44,  55,  69,  70,  74,  46,  50,
        46,  52,  46,   0,  57,  47,  47,  49,  58,  40,  94,  41,  43,
       105,  48,  54, 140, 144, 116, 164, 158,  35,  49,  36,  36,  41,
        57,  87,  73, 118,  58, 101,  70,  82, 106, 116,  80,  79,  67,
        93,  78,  72,  92,  75,  77,  59,  72,  46,  72,  78,  76,  84,
        71,  72,  57,  77, 110,  80,  75,  94,  73,  73,  93,  91,  95,
        75,  51,  41,  75,  77,  47,  41,  40,  41,  86,  89,  81,  78,
        74,  37,  44,  41,  43,  42,  41,  45,  56,  41,  52, 147,  44,
        80, 201,  35,  41,  55, 128,  38,  52,  48,  39,  70,  44,  45,
        38,  51,  19,  79,  55, 158,  40,  85, 121,  30, 111,  69,  78,
        46,  49,  51,  84,  84,  72,  75,  81, 117,  56,  75,  71,  90,
        78,  71,  81,  80,  69,  90,  67,  56,  86,  72,  75,  80,  41,
        44,  43,  70,  81,  71,  88,  44,  69,  46,  41,  55,  43,  48,
        42,  73,  87,  69,  76,  60,  47,  46,  44,  48,  74,  70,  92,
        75,  46,  39,  57,  92,  83,  81,  50,  50,  51,  40,  51,  89,
        75,  44,  43,  65,  43,  82,  56,  34,  61,  82,  84,  62,  80,
        46,  60,  42,  33,  75,  78,  54,  74,  39,  75,  69,  75,  73,
        70,  84,  79,  70,  78,  90,  81,  75,  69,  66,  75,  67,  76,
        67,  70,  79,  79,  71,  79,  83,  64,  75,  74,  81,  74,  77,
        67,  74,  73, 122,  82,  72,  41,  50,  41,  36,  50,  70,  59,
       176,  69,  45,  42,  65,  72,  41,  46,  48,  61,  55,  42,  49,
        43,  39,  44,  46,  39,  39,  47,  39,  39,  39,  36,  42,  47,
        66,  44,  42,  42,  49,  42,  41,  42,  47,   2,  10,   0,   1,
         4,   0,   0,   2,   3,  10,   1,   1,  17,   0,   6,  35,   7])

output_file('test.html')
fig=Histogram(a, bins=10, webgl=False)
show(fig)

print(np.histogram(a, bins=10))
```
The numpy histogram shows that the last bin count should be 1, whereas the displayed plat shows a value of 200. Clearly, from the figure, the sum of the y values is greater than the size (=429) of array a. That surely can't be correct.

Also attached is the image of the plot.
0.11.1                   py35_0  defaults 
![test](https://cloud.githubusercontent.com/assets/1182613/15351691/5e9a1ab2-1c95-11e6-85c7-1045ae7672a1.png)

**Comment by waqarmalik** [comment]
Added comparison of 'Charts'-based `Histogram` with the histogram generated using numpy and basic plotting.
```
import numpy as np
import bokeh.plotting as plt
from bokeh.charts import Histogram, output_file

a = np.array([  0,  89,  60,   0,  62,  76,   0,  62,   0,  70,   0,   0,   0,
        77,   0,  53,  67,  49,   0,   0,  61,   0,  55,  68,   0, 102,
        60,   0,  42,   0,   0,   0,   0,  66,   0,   5,  82,  64,  46,
        61,   0,   0,   0,  96,  89,   0,  74,   0,   0,  45,   0,   0,
        22,  55,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,
         0,   8,   0,   0,   0,   0,   0,   0,  26,   0,  37,   0,   0,
         0,   0,   0,   0,   0,   0,   0,   0,   9,   0,   0,  77,   0,
        61,   0,   0,  15,  12,  53,   0,   0,   0,  73,  68,  75,  71,
        68,  42,  50,  44,  37,  42,  40,  51,  59,  88,  43,  83,  78,
        80,  85,  48,  43,  47,  51,  44,  55,  69,  70,  74,  46,  50,
        46,  52,  46,   0,  57,  47,  47,  49,  58,  40,  94,  41,  43,
       105,  48,  54, 140, 144, 116, 164, 158,  35,  49,  36,  36,  41,
        57,  87,  73, 118,  58, 101,  70,  82, 106, 116,  80,  79,  67,
        93,  78,  72,  92,  75,  77,  59,  72,  46,  72,  78,  76,  84,
        71,  72,  57,  77, 110,  80,  75,  94,  73,  73,  93,  91,  95,
        75,  51,  41,  75,  77,  47,  41,  40,  41,  86,  89,  81,  78,
        74,  37,  44,  41,  43,  42,  41,  45,  56,  41,  52, 147,  44,
        80, 201,  35,  41,  55, 128,  38,  52,  48,  39,  70,  44,  45,
        38,  51,  19,  79,  55, 158,  40,  85, 121,  30, 111,  69,  78,
        46,  49,  51,  84,  84,  72,  75,  81, 117,  56,  75,  71,  90,
        78,  71,  81,  80,  69,  90,  67,  56,  86,  72,  75,  80,  41,
        44,  43,  70,  81,  71,  88,  44,  69,  46,  41,  55,  43,  48,
        42,  73,  87,  69,  76,  60,  47,  46,  44,  48,  74,  70,  92,
        75,  46,  39,  57,  92,  83,  81,  50,  50,  51,  40,  51,  89,
        75,  44,  43,  65,  43,  82,  56,  34,  61,  82,  84,  62,  80,
        46,  60,  42,  33,  75,  78,  54,  74,  39,  75,  69,  75,  73,
        70,  84,  79,  70,  78,  90,  81,  75,  69,  66,  75,  67,  76,
        67,  70,  79,  79,  71,  79,  83,  64,  75,  74,  81,  74,  77,
        67,  74,  73, 122,  82,  72,  41,  50,  41,  36,  50,  70,  59,
       176,  69,  45,  42,  65,  72,  41,  46,  48,  61,  55,  42,  49,
        43,  39,  44,  46,  39,  39,  47,  39,  39,  39,  36,  42,  47,
        66,  44,  42,  42,  49,  42,  41,  42,  47,   2,  10,   0,   1,
         4,   0,   0,   2,   3,  10,   1,   1,  17,   0,   6,  35,   7])

output_file('test.html')
p1=Histogram(a, bins=10, webgl=False, title='Using charts')

hist, edges = np.histogram(a, bins=10)
p2 = plt.figure(title='Using numpy', y_range=p1.y_range)
p2.quad(top=hist, bottom=0, left=edges[:-1], right=edges[1:], fill_color='red', 
        fill_alpha=0.7, line_color='black')

plt.show(plt.hplot(p1,p2))
```
![test2](https://cloud.githubusercontent.com/assets/1182613/15352465/337d747e-1c99-11e6-8c6e-6142fb6e4819.png)

**Comment by waqarmalik** [comment]
Search shows this to be duplicate of #3875, addressed by PR #3988
